pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -448,64 +447,44 @@ func (c *Reconciler) updateLabels(tr *v1alpha1.TaskRun) (*v1alpha1.TaskRun, erro
 // volumeMount
 func (c *Reconciler) createBuildPod(tr *v1alpha1.TaskRun, ts *v1alpha1.TaskSpec, taskName string) (*corev1.Pod, error) {
-	s := &buildv1alpha1.BuildSpec{
-		Steps:   ts.Steps,
-		Volumes: ts.Volumes,
-	}
-
-	bSpec := s.DeepCopy()
-	bSpec.Timeout = tr.Spec.Timeout
-	bSpec.Affinity = tr.Spec.Affinity
-	bSpec.NodeSelector = tr.Spec.NodeSelector
-
-	build := &buildv1alpha1.Build{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      tr.Name,
-			Namespace: tr.Namespace,
-			OwnerReferences: []metav1.OwnerReference{
-				*metav1.NewControllerRef(tr, groupVersionKind),
-			},
-			// Attach new label and pass taskrun labels to build
-			Labels: makeLabels(tr),
-		},
-		Spec: *bSpec,
-	}
-
-	build, err := resources.AddInputResource(c.KubeClientSet, build, taskName, ts, tr, c.resourceLister, c.Logger)
+	ts = ts.DeepCopy()
+	ts, err := resources.AddInputResource(c.KubeClientSet, taskName, ts, tr, c.resourceLister, c.Logger)
 	if err != nil {
 		c.Logger.Errorf("Failed to create a build for taskrun: %s due to input resource error %v", tr.Name, err)
 		return nil, err
 	}
 
-	err = resources.AddOutputResources(c.KubeClientSet, build, taskName, ts, tr, c.resourceLister, c.Logger)
+	err = resources.AddOutputResources(c.KubeClientSet, taskName, ts, tr, c.resourceLister, c.Logger)
 	if err != nil {
 		c.Logger.Errorf("Failed to create a build for taskrun: %s due to output resource error %v", tr.Name, err)
 		return nil, err
 	}
 
-	build, err = createRedirectedBuild(c.KubeClientSet, build, tr, c.cache, c.Logger)
+	fmt.Println("b:steps", len(ts.Steps), ts.Steps)
+	ts, err = createRedirectedBuild(c.KubeClientSet, ts, tr, c.cache, c.Logger)
 	if err != nil {
 		return nil, fmt.Errorf("couldn't create redirected Build: %v", err)
 	}
+	fmt.Println("a:steps", len(ts.Steps), ts.Steps)
 
 	var defaults []v1alpha1.TaskParam
 	if ts.Inputs != nil {
 		defaults = append(defaults, ts.Inputs.Params...)
 	}
 	// Apply parameter templating from the taskrun.
-	build = resources.ApplyParameters(build, tr, defaults...)
+	ts = resources.ApplyParameters(ts, tr, defaults...)
 
 	// Apply bound resource templating from the taskrun.
-	build, err = resources.ApplyResources(build, tr.Spec.Inputs.Resources, c.resourceLister.PipelineResources(tr.Namespace).Get, "inputs")
+	ts, err = resources.ApplyResources(ts, tr.Spec.Inputs.Resources, c.resourceLister.PipelineResources(tr.Namespace).Get, "inputs")
 	if err != nil {
 		return nil, fmt.Errorf("couldnt apply input resource templating: %s", err)
 	}
-	build, err = resources.ApplyResources(build, tr.Spec.Outputs.Resources, c.resourceLister.PipelineResources(tr.Namespace).Get, "outputs")
+	ts, err = resources.ApplyResources(ts, tr.Spec.Outputs.Resources, c.resourceLister.PipelineResources(tr.Namespace).Get, "outputs")
 	if err != nil {
 		return nil, fmt.Errorf("couldnt apply output resource templating: %s", err)
 	}
 
-	pod, err := resources.MakePod(build, c.KubeClientSet)
+	pod, err := resources.MakePod(tr, *ts, c.KubeClientSet)
```

Comment:
minor note, seems kinda too bad we have to pass around the `tr` and the `ts` everywhere - might be nice if there was some kind of intermediate object that held the data we need from both :D

Reply:
yeah, I was thinking about that. wasn't sure I should go through it in this PR or later on :sweat_smile:

Reply:
kk sounds good :D